File: frontriavett/src/types/appointment.ts
/**
 * Tipos para el módulo de Citas/Agenda
 */

import { VeterinarianResponse } from './veterinarian';

export type AppointmentStatus = 'PENDING' | 'CONFIRMED' | 'CANCELED' | 'COMPLETED';

export interface AppointmentRequest {
  patientId: string;
  veterinarianId: string;
  scheduledAt: string; // ISO 8601 format: 2025-11-25T10:00:00
}

export interface AppointmentResponse {
  id: string;
  patientId: string;
  veterinarianId: string;
  veterinarian?: VeterinarianResponse; // Información completa del veterinario
  scheduledAt: string;
  status: AppointmentStatus;
  createdAt: string;
  updatedAt: string;
}

export interface GetAppointmentsParams {
  veterinarianId?: string;
}

export interface AppointmentUpdateRequest {
  patientId?: string;
  veterinarianId?: string;
  scheduledAt?: string;
  status?: AppointmentStatus;
}

File: agendaservice/src/main/java/com/riavet/agendaservice/domain/model/Appointment.java
package com.riavet.agendaservice.domain.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "appointments")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(columnDefinition = "UUID")
    private UUID id;

    @Column(name = "patient_id", nullable = false, columnDefinition = "UUID")
    private UUID patientId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "veterinarian_id", nullable = false)
    private Veterinarian veterinarian;

    @Column(name = "scheduled_at", nullable = false)
    private LocalDateTime scheduledAt;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AppointmentStatus status;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    public enum AppointmentStatus {
        PENDING,
        CONFIRMED,
        COMPLETED,
        CANCELED
    }
}
